check_haproxy.rb

```diff
@@ -85,16 +91,28 @@
   exit UNKNOWN
 end
 
+
+tries = 2
+
 begin
   f = open(options.url, :http_basic_authentication => [options.user, options.password])
 rescue OpenURI::HTTPError => e
   puts "ERROR: #{e.message}"
-  exit UNKNOWN
+  exit CRITICAL
 rescue Errno::ECONNREFUSED => e
   puts "ERROR: #{e.message}"
-  exit UNKNOWN
+  exit CRITICAL
+rescue Exception => e
+  if e.message =~ /redirection forbidden/
```

Comment:
open-uri will follow redirects unless explicitly told not to, so this shouldn't be needed.  From https://twin.github.io/improving-open-uri/ this is only to limit the number of times open-uri attempts to redirect.

A bigger security issue is calling open directly and not parsing the uri, which needs to be fixed.